lib/logstash/inputs/s3.rb

```diff
@@ -257,6 +262,13 @@ def read_gzip_file(filename, block)
     File.open(filename) do |zio|
       while true do
         io = Zlib::GzipReader.new(zio)
+        case @gzip_prefer
+        when "memory"
+          io.each_line(&block)
+        when "speed"
+          io.read.each_line(&block)
+        else fail("invalid gzip_optimise configuration")
+        end
         io.each_line { |line| block.call(line) }
```

Comment:
Doesn't this `io.each_line` need to get removed?

Reply:
it does; however, I'm going to close this in favor of #127, which solves the same problem without adding confusing configuration.

Reply:
Ahh, nice. Thank you.